src/Microsoft.AspNet.Authentication.OpenIdConnect/OpenIdConnectAuthenticationHandler.cs

```diff
@@ -520,6 +540,14 @@ protected override AuthenticationTicket AuthenticateCore()
             }
         }
 
+        protected virtual IdentityModel.Clients.ActiveDirectory.AuthenticationResult RedeemAuthorizationCode(string authorizationCode)
+        {
+            AuthenticationContext authContext = new AuthenticationContext(Options.Authority, false);
+            ClientCredential credential = new ClientCredential(Options.ClientId, Options.ClientSecret);
+            var tokens = authContext.AcquireTokenByAuthorizationCodeAsync(authorizationCode, new Uri(Options.RedirectUri), credential).GetAwaiter().GetResult();
```

Comment:
And BTW, given that `Options.RedirectUri` is not a mandatory setting, `Uri`'s constructor will throw an exception if you don't set it.